pkg/gitparse/gitparse.go

```diff
@@ -235,6 +235,10 @@ func (c *Parser) RepoPath(
 		"log",
 		"--patch", // https://git-scm.com/docs/git-log#Documentation/git-log.txt---patch
 		"--full-history",
+		// Limit ourselves to a single line of context.  This context is necessary to include the match key for
+		// matching detectors when the match string isn't on the same line as the secret, and the match key
+		// previously existed (as when updating private key certs)
+		"--unified=1", // https://git-scm.com/docs/git-log#Documentation/git-log.txt-code--unifiedltngtcode
```

Comment:
Won't this introduce the risk of accidentally associating a finding with the wrong commit? E.g:
```
MYTOKEN=examplesecret
+changedline
```